lib/network.go

```diff
@@ -6391,70 +6455,116 @@ type MessagingGroupMetadata struct {
+	ret.AccessPublicKey, err = ReadVarString(rr)
 	if err != nil {
-		return errors.Wrapf(err, "MessagingGroupMetadata.FromBytes: "+
-			"Problem reading MessagingPublicKey")
+		return errors.Wrapf(err, "AccessGroupMetadata.FromBytes: "+
+			"Problem reading AccessPublicKey")
 	}
 
-	ret.MessagingGroupKeyName, err = ReadVarString(rr)
+	ret.AccessGroupKeyName, err = ReadVarString(rr)
 	if err != nil {
-		return errors.Wrapf(err, "MessagingGroupMetadata.FromBytes: "+
-			"Problem reading MessagingGroupKey")
+		return errors.Wrapf(err, "AccessGroupMetadata.FromBytes: "+
+			"Problem reading AccessGroupKey")
 	}
 
 	ret.GroupOwnerSignature, err = ReadVarString(rr)
 	if err != nil {
-		return errors.Wrapf(err, "MessagingGroupMetadata.FromBytes: "+
+		return errors.Wrapf(err, "AccessGroupMetadata.FromBytes: "+
 			"Problem reading GroupOwnerSignature")
 	}
 
 	numRecipients, err := ReadUvarint(rr)
 	for ; numRecipients > 0; numRecipients-- {
-		recipient := &MessagingGroupMember{}
+		recipient := &AccessGroupMember{}
 		if err := recipient.FromBytes(rr); err != nil {
-			return errors.Wrapf(err, "MessagingGroupMetadata.FromBytes: "+
+			return errors.Wrapf(err, "AccessGroupMetadata.FromBytes: "+
 				"error reading recipient")
 		}
-		ret.MessagingGroupMembers = append(ret.MessagingGroupMembers, recipient)
+		ret.AccessGroupMembers = append(ret.AccessGroupMembers, recipient)
 	}
 
 	*txnData = ret
 	return nil
 }
 
-func (txnData *MessagingGroupMetadata) New() DeSoTxnMetadata {
-	return &MessagingGroupMetadata{}
+func (txnData *AccessGroupMetadata) New() DeSoTxnMetadata {
+	return &AccessGroupMetadata{}
+}
+
+// GetAccessGroupOperation returns the messaging group operation based on the transaction. In particular, we check
+// transaction's ExtraData to see whether it's a mute/unmute operation.
+func GetAccessGroupOperation(txn *MsgDeSoTxn) (AccessGroupOperation, error) {
+	// If the transaction is nil then we return an error.
+	if txn == nil {
+		return 0, fmt.Errorf("GetAccessGroupOperation: nil txn")
+	}
+
+	// Make sure this is a messaging group transaction.
+	if txn.TxnMeta.GetTxnType() != TxnTypeAccessGroupCreate {
+		return 0, fmt.Errorf("GetAccessGroupOperation: called on txn with type %v", txn.TxnMeta.GetTxnType())
+	}
+
+	// Sanity-check cast the transaction metadata to a messaging group metadata.
+	_, ok := txn.TxnMeta.(*AccessGroupMetadata)
+	if !ok {
+		return 0, fmt.Errorf("GetAccessGroupOperation: called on txn with type %v", txn.TxnMeta.GetTxnType())
+	}
+
+	// If the transaction's ExtraData is nil then we assume AccessGroupOperationAddMembers
+	messagingGroupOperation := AccessGroupOperationAddMembers
+	if txn.ExtraData == nil {
+		return messagingGroupOperation, nil
+	}
+
+	// Check if the transaction's ExtraData contains a AccessGroupOperationType.
+	if operationTypeBytes, operationTypeExists := txn.ExtraData[AccessGroupOperationType]; operationTypeExists && len(operationTypeBytes) == 1 {
```

Comment:
we should probably through some sort of error if the length of opeationTypeBytes is greater than expected

Reply:
Thanks, done. Now throwing error if len(operationTypeBytes) != 1